### PR TITLE
feat/events-incoming

### PR DESCRIPTION
**Pull request by jeremiebardon**
Register button visibility & Show event incoming.

### CODE REVIEW

**Comment by jeremiebardon** [comment]
Maybe we can seperate the list in 2: 

- First incomming events: add a header and show events below. If no event give a link to create on or share.

- Second the past events with an header also 

ExplanationInImage

**Comment by f3ltron** [comment]
J'aime beaucoup !

File: configs/modules.js [hunk 1]
```diff
@@ -1,3 +1,39 @@
+<<<<<<< HEAD
```
Comment (f3ltron) [inline configs/modules.js:1]: you didnt correct all merge conflict

File: .babelrc [hunk 2]
```diff
@@ -1,16 +1,16 @@
-{
-  "env": {
```
Comment (f3ltron) [inline .babelrc:2]: can you just push all your modificatiion without the merge files ?

**Comment by f3ltron** [comment]
I commented if you can do it and after i will merge it.

I have created a test futur event for you to test too

**Comment by f3ltron** [comment]
If you can do the separation i will be ok with that. Just maybe a simple green square is ok no need of the border multi green. @aurelienroux  what do you think ?

After you two last ticket i will close 2.0 and start the project 2.1

**Comment by f3ltron** [comment]
For the next ticket jeremie you will be able to work directly on vuemontreal github project no need to fork anymore. It will be easier to check your work 👍

**Comment by jeremiebardon** [comment]
Ok! Just to notice: I've push the resoloved conflict version. I'm going to accept your invit! Thank you :)

**Comment by aurelienroux** [comment]
> If you can do the separation i will be ok with that. Just maybe a simple green square is ok no need of the border multi green. @aurelienroux what do you think ?
> 
> After you two last ticket i will close 2.0 and start the project 2.1



> If you can do the separation i will be ok with that. Just maybe a simple green square is ok no need of the border multi green. @aurelienroux what do you think ?
> 
> After you two last ticket i will close 2.0 and start the project 2.1

@f3ltron @jeremiebardon For some reasons, I can't preview the branch and pr directly. I think it's because it's coming from a fork but not sure... 
But in any case, I agree with Florent, let's stick with the original design and colors so no gradient on the image and just a simple border with the theme green color.

**Comment by f3ltron** [comment]
Yes you have to do special commands or just clone his repo

**Comment by aurelienroux** [comment]
ok got it. Yeah definitly remove the color gradient. Also, I think the section headers **upcoming events / past events** were a nice idea and should be included as well. WDYT ?

**Comment by jeremiebardon** [comment]
> ok got it. Yeah definitly remove the color gradient. Also, I think the section headers **upcoming events / past events** were a nice idea and should be included as well. WDYT ?

You can test, it's live on this pull request :) About the method used, what do you think?

**Comment by aurelienroux** [comment]
> You can test, it's live on this pull request :) About the method used, what do you think?

Yep, I see it. I have 2 points on this

First, I don't think the argument on the **isIncoming** function should be called **event** as it is confusing with the component **event prop**. IMO, it should be called **eventSortDate** or something like this to clarify that we are working with a date string and not the props object.

Second, the function is well done but I think you may have a problem on the timestamp created (but I'm not sure so it's best if @f3ltron can confirm what I say). 
Reason for that is the **event.sortByDate** string is in this format 2020-05-07, meaning you have no hour, minutes or seconds info, resulting in the timestamp conversion as 1588809600000 and an exact hour of GMT: Thursday 7 May 2020 00:00:00 (so midnight basically). 
But the **Date.now()** will return the exact hour, meaning that on the day of the event at midnight, the event will already be classified as past event, even if it will be happening on the future evening. 

WDYT ? maybe I'm seeing a future bug where there is none, it's totally possible

File: components/eventPreview.vue [hunk 3]
```diff
@@ -1,80 +1,89 @@
...
-}
-
-.button-blue {
-  @apply bg-mtl-accent;
-}
-</style>
+<template>
+  <article
+    :class="
+      isIncomming
```
Comment (aurelienroux) [inline components/eventPreview.vue:4]: isIncoming has only one m ;)

File: components/eventPreview.vue [hunk 4]
```diff
@@ -1,80 +1,89 @@
...
+          </li>
+        </ul>
+      </div>
+    </header>
+    <main class="my-12">
+      <text-description :text="event.content.description" />
+    </main>
+    <footer class="border-t py-6 text-right">
+      <a
+        v-if="event.content.inscription.url && isIncomming"
```
Comment (aurelienroux) [inline components/eventPreview.vue:45]: same here

File: components/eventPreview.vue [hunk 5]
```diff
@@ -1,80 +1,89 @@
...
+export default {
+  components: {
+    TextDescription
+  },
+  props: {
+    event: {
+      type: Object,
+      required: true
+    },
+    isIncomming: {
```
Comment (aurelienroux) [inline components/eventPreview.vue:70]: same here

File: components/eventPreview.vue (hunk 3 above)
Comment (jeremiebardon) [reply components/eventPreview.vue]: Ok for this one

**Comment by f3ltron** [comment]
@aurelienroux when you think it's done accept the pr thank's for all your work :) so excited to start the 2.1 :).

**Comment by aurelienroux** [comment]
> Maybe the second one is better because (I think) the event start at a specific hour exact ?

Maybe try something like this 
https://stackoverflow.com/questions/2698725/comparing-date-part-only-without-comparing-time-in-javascript
When you re getting the Date.Now() object, just set the hours to 23:59:59

Or as you say, just add one day directly in the comparison operator. this should do the trick

**Comment by f3ltron** [comment]
and good news now i will be able to make auto deployment for validate branch on netlify

File: pages/index.vue [hunk 6]
```diff
@@ -113,11 +113,11 @@ export default {
   methods: {
     pastEvents: (events) =>
       events.filter(
-        (event) => new Date(event.sort_by_date).getTime() < Date.now()
+        (event) => new Date(event.sort_by_date).setHours(24) < Date.now()
```
Comment (aurelienroux) [inline pages/index.vue:115]: Nice

**Comment by tlebel** [comment]
@jeremiebardon @f3ltron Good job guys on the website :) 

I just noticed here the changeset is humongous though! 
Basically, every lines of every files touched are "changed". 

The problem is that @jeremiebardon committed files with CRLF  line ending. Ironically, we already have a protection against it here : https://github.com/vuemontreal/vuemontreal/blob/928991b80e8cdce6bd76cba9e95f2ec6936a158b/.editorconfig#L7

I see two possibilities: 

1. your code editor does not have support for `.editorconfig` files
1. your git (most probably installed on Windows) has the CRLF settings `core.autocrlf` set to `false`. Have a look here it tells you how to fix that for next time: https://stackoverflow.com/a/20653073/1375487

The trick is also to watch for it at the bottom of your editor. Normally, it shows the current line ending: 

All @jeremiebardon files now have: 


Previous files all have:

**Pull request merged**